src/js/notifications-sw.js

```diff
@@ -0,0 +1,76 @@
+import { default as localforage } from 'localforage';
+
+self.addEventListener('push', event => {
+  function getPayload() {
+    if (event.data) {
+      return Promise.resolve(event.data.json());
+    }
+    return localforage.getItem('deviceId')
+    .then(deviceId => {
+      if (!deviceId) {
+        return null;
+      }
+      return fetch(`./payload/${deviceId}`)
+      .then(response => response.json());
+    });
+  }
+
+  event.waitUntil(
+    getPayload()
+    .then(data => {
+      // TODO add a way to unregister from the back-end
+      // data.command (?)
+      const title = data ? data.title : 'Platform Status';
+      const body = data ? data.body : 'Notification Error';
```

Comment:
I think it'd be better to not show errors as notifications. So, if there isn't a body, just return early.